portal-impl/src/com/liferay/portlet/asset/service/impl/AssetCategoryLocalServiceImpl.java

```diff
@@ -424,6 +427,48 @@ public List<AssetCategory> getEntryCategories(long entryId) {
 		return Collections.emptyList();
 	}
 
+	@Override
+	public AssetCategory getOrAddIncompleteCategory(
+			String externalReferenceCode, long userId, long groupId,
+			long parentCategoryId, Map<Locale, String> titleMap,
+			Map<Locale, String> descriptionMap, long vocabularyId,
+			String[] categoryProperties)
```

Comment:
Is it correct that ALL of these parameters are needed to create an incomplete category? If I'm able to provide them at this point why I cannot create a "complete" one?

I think these should be the minimum set of data needed to create the "row" - as I can see they are also all updatable with the update method (so are not unmodifiable attributes - a part of `String externalReferenceCode, long userId, long groupId`) that should be there as also needed to do the fetch

Reply:
yeah, I was delegating the creation of the "not needed parameters" to the place the method is called, but to simplify the calls, I can make the creation of them here on the method.

I will change them

Reply:
and is the name strictly needed or could be optional? like if it's null or empty we use the erc as name?
Because my "problem" here is that having the vocabularyId as parameter it means that "before" I have to create it (so the order of import/creation returns to be relevant and is something that we are trying to remove with this lazy behavior)

Reply:
About the name / type
 it's the same pattern than on account or Role. Should we change that too?

The Vocabulary/parent category 
I will talk with the team about options because we see the need of a hierarchy.


// cc @robertoDiaz

Reply:
Hi @smottal 

I changed some things as we talked about. If you please can review this changes again I'll be grateful